commit b6fa3778e84c9d2f6d9511189ba16078b6c37196
Author: Rudi Heitbaum <rudi@heitbaum.com>
Date:   Mon Oct 17 11:40:02 2022 +0000

    Revert "fbdev: Make registered_fb[] private to fbmem.c"
    
    This reverts commit 5727dcfd8486399c40e39d2c08fe36fedab29d99.

diff --git a/drivers/video/fbdev/core/fbmem.c b/drivers/video/fbdev/core/fbmem.c
index 1e70d8c67653..6ae1c5fa19f9 100644
--- a/drivers/video/fbdev/core/fbmem.c
+++ b/drivers/video/fbdev/core/fbmem.c
@@ -51,10 +51,10 @@
 static DEFINE_MUTEX(registration_lock);
 
 struct fb_info *registered_fb[FB_MAX] __read_mostly;
+EXPORT_SYMBOL(registered_fb);
+
 int num_registered_fb __read_mostly;
-#define for_each_registered_fb(i)		\
-	for (i = 0; i < FB_MAX; i++)		\
-		if (!registered_fb[i]) {} else
+EXPORT_SYMBOL(num_registered_fb);
 
 bool fb_center_logo __read_mostly;
 
diff --git a/include/linux/fb.h b/include/linux/fb.h
index 0aff76bcbb00..453c3b2b6b8e 100644
--- a/include/linux/fb.h
+++ b/include/linux/fb.h
@@ -627,10 +627,16 @@ extern int fb_get_color_depth(struct fb_var_screeninfo *var,
 extern int fb_get_options(const char *name, char **option);
 extern int fb_new_modelist(struct fb_info *info);
 
+extern struct fb_info *registered_fb[FB_MAX];
+extern int num_registered_fb;
 extern bool fb_center_logo;
 extern int fb_logo_count;
 extern struct class *fb_class;
 
+#define for_each_registered_fb(i)		\
+	for (i = 0; i < FB_MAX; i++)		\
+		if (!registered_fb[i]) {} else
+
 static inline void lock_fb_info(struct fb_info *info)
 {
 	mutex_lock(&info->lock);
